Expensify/App
# Move create temp login logic to it's own function

Call createTemporaryLogin in the set password function to allow token refresh after exp
Change redirect check to look for newly named generated password key in ONYX

### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1482

### Tests

1. Trigger password reset on expensify.com for your account.
2. Make sure you are logged out.
3. Navigate to: http://expensify.cash/#/setpassword/[code from magic link in email here]
4. Make sure you have credentials.login key set to your email address in local storage.
5. Set a new password.
6. Edit the session.authtoken in local storage and deface it somehow.
7. if you have any photos in the chat log, they will disappear.
8. Do any action that initiates an API call.
9. Check and see that the auth token has been replaced and the app works normally.
10. Navigate to http://expensify.cash/#/setpassword/[any junk here]
11. See how you're sent to the root of the app.

This time follow steps 6-11 but after logging in normally to ensure previous functionality is still working.




### Tested On


### Screenshots
I didn't touch any presentational logic or stylings, there isn't anything to screenshot.
